src/Database/PostgreSQL/Protocol/DataRows.hs

```diff
@@ -183,8 +182,8 @@ reverseDataRows :: DataRows -> DataRows
 reverseDataRows = foldlDataRows (flip chunk) Empty
 
 {-# INLINE countDataRows #-}
-countDataRows :: DataRows -> Word
-countDataRows = foldlDataRows (\acc (DataChunk c _) -> acc + c) 0
+countDataRows :: [B.ByteString] -> Int
+countDataRows = length
```

Comment:
It seems that the logic changed there, are we guaranteed that there is one row per list element?